test/unit/testSignOut.js

```diff
@@ -0,0 +1,40 @@
+const initSteps = require('app/core/initSteps');
+const assert = require('chai').assert;
+const sinon = require('sinon');
+const when = require('when');
+const co = require('co');
+const services = require('app/components/services');
+
+describe('Sign-Out', function () {
+
+    const steps = initSteps([`${__dirname}/../../app/steps/action/`, `${__dirname}/../../app/steps/ui`]);
+
+    it('test authToken, cookies and session data have been removed from the session', () => {
+        const signOut = steps.SignOut;
+        const signOutStub = sinon.stub(services, 'signOut');
+        signOutStub.returns(when(200));
+
+        const req = {
+            cookies: {
+                _ga: 'dummy_ga',
+                _gid: 'dummy_gid',
+                _gat: '1'
+            },
+            session: {
+                form: {
+                    payloadVersion: '4.1.0',
+                    applicantEmail: 'test@email.com'
+                },
+                destroy: () => {
+                    delete this.session;
+                }
+            },
+        };
+
+        co(function* () {
+            yield signOut.getContextData(req);
+            assert.isNull(req.cookies);
+            assert.isNull(req.session);
+        });
```

Comment:
If you comment out either of the `delete req.cookies;` or `req.session.destroy();` lines then the tests don't fail.

- I don't think we need to use `co` here but do this with `.then()`. 
- We can use `sinon.spy()` to spy on the req to check if the values have been removed.
- It also looks like we don't need to mock `req.session.destroy()`.

Now when you comment out either of the delete/destroy lines above the tests fail.

For example:

```
const initSteps = require('app/core/initSteps');
const expect = require('chai').expect;
const sinon = require('sinon');
const when = require('when');
const services = require('app/components/services');

describe.only('Sign-Out', function () {
    const steps = initSteps([`${__dirname}/../../app/steps/action/`, `${__dirname}/../../app/steps/ui`]);

    it('test authToken, cookies and session data have been removed from the session', (done) => {
        const signOut = steps.SignOut;
        const signOutStub = sinon.stub(services, 'signOut');
        signOutStub.returns(when(200));

        const req = {
            cookies: {
                _ga: 'dummy_ga',
                _gid: 'dummy_gid',
                _gat: '1'
            },
            session: {
                form: {
                    payloadVersion: '4.1.0',
                    applicantEmail: 'test@email.com'
                }
            },
        };

        req.session.destroy = sinon.spy();
        req.cookies = sinon.spy();

        signOut.getContextData(req).then(() => {
            expect(req.cookies).to.equal(undefined);
            expect(req.session.destroy).to.have.been.calledOn(req.session);
            done();
        });
    });
});
```